crates/pypi-types/src/base_url.rs

```diff
@@ -36,10 +39,12 @@ pub fn base_url_join_relative(base: &str, maybe_relative: &str) -> Result<Url, J
 /// The error message includes the URL (`base` or `maybe_relative`) passed to
 /// `base_url_join_relative` that provoked the error.
 #[derive(Clone, Debug, thiserror::Error)]
-#[error("Failed to parse URL: `{original}`")]
-pub struct JoinRelativeError {
-    original: String,
-    source: url::ParseError,
+pub enum JoinRelativeError {
+    #[error("Failed to parse URL: `{original}`")]
+    ParseError {
+        original: String,
+        source: url::ParseError,
+    },
 }
```

Comment:
I changed this when I was handling another error kind in here. Think it makes sense to leave it for the future, I prefer enum error wrappers.